### PR TITLE
add uuid as fingerprint data

### PR DESCRIPTION
**Pull request by sparrowDom**
### Description:

Create an UUID and populate it with first `walletId` that tries to Enrol into growth (or use `withFingerprint` hoc). Store that UUID in `localStorage` and a `cookie`

In future it would be even more resilient storing this into IndexedDB, but that is not a part of this PR. 

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: dapps/marketplace/src/hoc/withFingerprint.js [hunk 1]
```diff
@@ -11,28 +14,60 @@ const browserPropsWhitelist = [
...
+  cachedFingerprintData.uuid = cachedUUID
+  storeUUID(cachedUUID)
+  return cachedFingerprintData
+}
+
 const getFingerprint = memoize(getFingerprintFn)
 
 function withFingerprint(WrappedComponent) {
   const WithFingerprint = props => {
+    if (!props.wallet) {
```
Comment (franckc) [inline dapps/marketplace/src/hoc/withFingerprint.js]: I wonder if we should still let the code execute but just not include the uuid in the fingerprint data ?

Reasoning being that we may use this WithFingerprint module in the future in other places where a wallet may not be accessible ?

File: dapps/marketplace/src/hoc/withFingerprint.js [hunk 2]
```diff
@@ -2,7 +2,10 @@ import React, { useState, useEffect } from 'react'
 import Fingerprint2 from 'fingerprintjs2'
 import memoize from 'lodash/memoize'
 
-let cachedFingerprintData
+import withWallet from './withWallet'
+
+let cachedFingerprintData, cachedUUID
+const uuidKey = 'rewards_uuid'
```
Comment (franckc) [inline dapps/marketplace/src/hoc/withFingerprint.js]: Great naming. uuid is a perfect name for this id.

File: dapps/marketplace/src/hoc/withFingerprint.js (hunk 1 above)
Comment (sparrowDom) [reply dapps/marketplace/src/hoc/withFingerprint.js]: That is a great point, changing to your suggestion

File: dapps/marketplace/src/hoc/withFingerprint.js (hunk 2 above)
Comment (sparrowDom) [reply dapps/marketplace/src/hoc/withFingerprint.js]: 😊

**Comment by franckc** [comment]
Nice. Thanks for including my suggestion. Feel free to merge when you are ready ! :)

**Comment by sparrowDom** [comment]
was waiting for tests, all green now

**Pull request merged**